### DISCUSSION TITLE
Inconsistent FRB designation (inaccurately-assigned trailing letter)

### DISCUSSION THREAD

**Discussion by 0xCoto** [chime-frb-open-data/community#23] [Q&A] [answered]
I've noticed an event which has a tiny inaccuracy in the designation of the FRB name.

## Expected Behavior
The TNS FRB designation follows the following rule: the trailing letter in the reported name (`A`, `B`, ...) shall be assigned based on the order of the time of the occurrence of the event. For example, the designation implies that the (hypothetical) `FRB 20201234A` burst occurred **_before_** `FRB 20201234B`.

## Current Behavior
The [CHIME/FRB catalog](http://chime-frb.ca/catalog) contains at least one instance where this designation format is not followed:

FRB | UTC | MJD | Telescope | RA | Dec. | _l_ | _b_ | Frequency | DM | Flux | Width | Fluence | S:N | Reference
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
FRB 20180814A | 2018-08-14 14:49:50 | 58344.61794 | CHIME | 4:22:00 | +73:36:00 | 136.46 | 16.58 | 464.2 | 190.9 | 0.4 | 7.86 | 3 | 11 | [Ref.](http://chime-frb.ca/catalog/FRB20180814A)
FRB 20180814B | 2018-08-14 14:20:17 | 58344.59742 | CHIME | 15:53:02.4 | +74:00:00 | 108.6 | 37.62 | 800.2 | 237.8 | 3 | 5.9 | 11 | 18.7 | [Ref.](http://chime-frb.ca/catalog/FRB20180814B)

Based on the catalog, `FRB 20180814A` occurred **_after_** `FRB 20180814B`, but the suffixes are obviously flipped.

## Possible Solution
I've only spotted one such instance with a very brief look, so hopefully this is not a frequently-appearing issue. Perhaps the simplest solution would be to manually apply any detected inconsistencies to the database.

## Context
I don't think this will have a very meaningful impact on my automation/analysis pipelines, so it's not a major concern for me at the moment, but it'd definitely be nice to maintain a dataset that is as consistent as possible. I am planning to apply such corrections to my database, but I believe that differing public datasets with very similar entries could potentially create unnecessary confusion to researchers.

I will report any other inconsistency I discover so that you're informed, and I'd be happy to somehow merge my corrections to the CHIME/FRB catalog page. Hope that helps!

**Comment by shinybrar** [comment, marked as answer]
@0xCoto You are absolutely right, we looked at the discrepancy but this is actually an expected behavior and could possible happen again in the future because of the way things are setup. See detailed explanation below:

1. When an FRB is detected, an autonomous pipeline sends the information about the FRB to the TNS nameservers which assigns it an official name. (Lets call this `FRBa`)
2. But if an FRB-like event that was detected before this so called `FRBA`, but was misclassified as an RFI event by our real-time pipelines,  we perform a second layer of checks which require human verification as well, and these re-classified FRBs can take days to weeks for us to ascertain if they are real. (Lets call this `FRBb`)
3. In this scenario, even if `FRBb` happened before `FRBa`, the TNS nameserver will not re-index the FRBs to assert that `FRBa` happened before `FRBb`.

In conclusion, we don''t expect the suffix to indicate the chronological order of the FRB's but rather the UTC time itself.
